### DISCUSSION TITLE
500 remote custom web buttons wont' show so cannot push button

### DISCUSSION THREAD

**Discussion by sp222222** [GerryDazoo/Slinger#26] [General]
more joy (and confusion) this one I've not messed with in 7 days yet I cannot get the remote webpage to show my custom findings (the crucial DVR LIST button) 

 : 3.04 Running on Windows-10-10.0.19044-SP0
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.1.88', 5224)
('192.168.1.88', 5224) OK
Using slingbox at  ('192.168.1.88', 5224)
Streamer:  500 downstairs Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from C:\Users\speed\Box\sling after 2022\remote_500.txt
Traceback (most recent call last):
  File "slingbox_server.py", line 1000, in <module>
  File "slingbox_server.py", line 955, in BuildPage
  File "slingbox_server.py", line 791, in parse_buttons
    def button():
ValueError: not enough values to unpack (expected 2, got 1)
[1908] Failed to execute script 'slingbox_server' due to unhandled exception!

**Comment by GerryDazoo** [comment]
You’ve probably fat fingered an entry in the remote definition file. I’m guessing a missing “:”. I’ll add some defensive code to catch this error and hopefully can print out the offending line.

In the mean time can you send me the contents of you remote config file so I can take a look. It might be a bug.

 

From: sp222222 ***@***.***> 
Sent: Monday, August 15, 2022 8:08 PM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] 500 remote custom web buttons wont' show so cannot push button (Discussion #26)

 

more joy (and confusion) this one I've not messed with in 7 days yet I cannot get the remote webpage to show my custom findings (the crucial DVR LIST button)

: 3.04 Running on Windows-10-10.0.19044-SP0
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running: 10 config.ini 8080
starting up on port 8080
Checking for slingbox at ('192.168.1.88', 5224)
('192.168.1.88', 5224) OK
Using slingbox at ('192.168.1.88', 5224)
Streamer: 500 downstairs Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from C:\Users\speed\Box\sling after 2022\remote_500.txt
Traceback (most recent call last):
File "slingbox_server.py", line 1000, in 
File "slingbox_server.py", line 955, in BuildPage
File "slingbox_server.py", line 791, in parse_buttons
def button():
ValueError: not enough values to unpack (expected 2, got 1)
[1908] Failed to execute script 'slingbox_server' due to unhandled exception!

**Reply by sp222222** [reply]
all my buttons work now - thanks to that debug comment on slingplayer 2.0 app it's for direct TV if anyone wants a copy , let me know genie model

**Comment by sp222222** [comment]
Thanks for getting this working for me Gerry. I added that line to the end of the config.ini file and it threw things off

**Comment by JoeM2016** [comment]
Any suggestions how to use the custom Remote definitions for the Tivo.Bolt.txt file.  I just downloaded it from the custom remote folder on the main page and included it in the folder with the other server files.  The remote section from my config file is at the end of this section.  I just get the default remote in a browser window.
 
Version : 3.07 Running on Windows-10-10.0.22000-SP0 pid= 18628
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.1.108', 5201)
**Reading Custom Remote definition from Tivo.Bolt.txt
Error reading remote definition file, using defaults**
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.1.108:8081 (Press CTRL+C to quit)
Error connecting to  ('192.168.1.108', 5201)
No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  fe80::3010:30ba:1a07:23a8%4
Finding Slingbox on local network. My IP Info =  fe80::64f9:e0bd:2e05:47bf%2
Finding Slingbox on local network. My IP Info =  fe80::f8f5:7c99:1d48:2376%10
Finding Slingbox on local network. My IP Info =  fe80::4c72:fd06:a014:5919%16
Finding Slingbox on local network. My IP Info =  fe80::cc6d:99de:d0e3:fb0%13
Finding Slingbox on local network. My IP Info =  192.168.1.108
Slingbox Found 192.168.1.171 5201 " My Slingbox                      "
Finding Slingbox on local network. My IP Info =  fe80::d5f0:ad19:7fbf:6427%20
Finding Slingbox on local network. My IP Info =  ::1
Finding Slingbox on local network. My IP Info =  127.0.0.1
Using slingbox at  ('192.168.1.171', 5201)
Streamer:   Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

from config.ini

[REMOTE]
; see release notes for changing this if your remote doesn't work
;code=0
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
include=Tivo.Bolt.txt

**Reply by GerryDazoo** [reply]
The error message is telling me the server can't find the  Tivo.Bolt.txt in the same folder as the slingbox_server. Copy it from the custom remotes folder or use the full path in the include=

**Reply by JoeM2016** [reply]
Thanks, it started working when I added the full path of the file location in the include line of the config file.
